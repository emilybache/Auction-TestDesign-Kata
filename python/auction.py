
class AuctionEventListener:
    # TODO: add methods here
    pass

    def auction_closed(self):
        pass

    def current_price(self, price):
        pass


class AuctionMessageTranslator:
    def __init__(self, listener: AuctionEventListener):
        self.listener = listener

    def process_message(self, message: str):
        if "CLOSE" in message:
            self.listener.auction_closed()
        elif "PRICE" in message:
            data = self.parse_message_data(message)

            current_price = int(data["CurrentPrice"])
            increment = int(data["Increment"])
            bidder = data["Bidder"]

            self.listener.current_price(current_price)
        else:
            # bug: should notify listener
            pass

    def parse_message_data(self, message):
        data = {}
        for element in message.split(";"):
            if not element:
                continue
            pair = element.split(":")
            data[pair[0].strip()] = pair[1].strip()
        return data

