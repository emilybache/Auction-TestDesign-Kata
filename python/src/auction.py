
class AuctionEventListener:
    # TODO: add methods here
    pass

class AuctionMessageTranslator:
    def __init__(self, listener: AuctionEventListener):
        self.listener = listener

    def process_message(self, message: str):
        if "CLOSE" in message:
            pass
            # bug: should notify listener
        elif "PRICE" in message:
            data = {}
            for element in message.split(";"):
                if not element:
                    continue
                pair = element.split(":")
                data[pair[0].strip()]  = pair[1].strip()

            current_price = int(data["CurrentPrice"])
            increment = int(data["Increment"])
            bidder = data["Bidder"]

            # bug: should notify listener
        else:
            # bug: should notify listener
            pass

