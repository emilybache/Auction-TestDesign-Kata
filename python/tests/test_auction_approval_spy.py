from approvaltests import verify

from auction import AuctionMessageTranslator, AuctionEventListener


class AuctionEventListenerApprovalSpy(AuctionEventListener):

    def __init__(self):
        self.log = ""

    def __repr__(self):
        return self.log

    def auction_closed(self):
        self.log += "auction_closed"

    def current_price(self, price):
        self.log += f"current_price({price})"


def test_notifies_auction_closed_when_close_message_received():
    message = "SOLVersion: 1.1; Event: CLOSE;"
    spy = AuctionEventListenerApprovalSpy()
    target = AuctionMessageTranslator(spy)

    target.process_message(message)

    verify(spy)


def test_notifies_bid_details_when_price_message_received():
    message = "SOLVersion: 1.1; Event: PRICE; CurrentPrice: 192; Increment: 7; Bidder: Someone else;"
    spy = AuctionEventListenerApprovalSpy()
    target = AuctionMessageTranslator(spy)

    target.process_message(message)

    verify(spy)
