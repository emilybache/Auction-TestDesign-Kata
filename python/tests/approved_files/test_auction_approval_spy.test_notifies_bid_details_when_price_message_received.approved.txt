current_price(192)
